integration-tests/maven/src/test/java/io/quarkus/maven/it/DevMojoIT.java

```diff
@@ -905,4 +904,16 @@ public void testThatDependencyInParentIsEvaluated() throws IOException, MavenInv
         testDir = initProject("projects/multimodule-parent-dep");
         runAndCheck();
     }
+
+    @Test
+    public void testThatGenerateCodeGoalIsNotTriggeredIfNotConfigured() throws IOException, MavenInvocationException {
+        testDir = initProject("projects/classic-no-generate");
+        // the skip parameter triggers a log statement by the generate goal,
+        // otherwise there would be no way to tell from the logfile that the goal was invoked
+        runAndCheck(false, "-Dquarkus.generate-code.skip=true");
+
+        assertThat(running.log()).doesNotContain("Skipping Quarkus code generation");
```

Comment:
Not ideal, but I did not find a better way to check whether the `generate-code` is actually invoked.